### ISSUE TITLE
Specification of files required for `read_10x_mtx`

### ISSUE THREAD

**Issue by WeilerP** [scverse/scanpy#2128]
## Description

In the docs for `scanpy.read_10x_mtx`, the mentioned file extensions and files are all uncompressed, _e.g._, `matrix.mtx`. However, when specifying a path which does not include the required file the `FileNotFoundError` calls it `matrix.mtx.gz`. For consistency, I think it'd be good to use either the compressed or uncompressed format in both cases, _i.e._, either update the docs or the `FileNotFoundError`. Otherwise, users might think they might have to compress the file (if there is a small typo in the provided path, for example).

**Comment by ivirshup** [comment]
We use whatever cellranger outputs. For cellranger 3.0.0+ those files are gzip compressed.

I would be fine with either of those changes. Could you make a PR?

More general note, I wouldn't want the function to look for either compressed or uncompressed. It should just be outputs from cellranger. This is to avoid scope creap – e.g. we aren't looking to support arbitrary "cellranger-like delimited files".

**Comment by WeilerP** [comment]
Yes, agreed. I was only talking about consistency between the raised error and docs. And yes, will open a PR.
